Makefile

```diff
@@ -574,12 +574,21 @@ misspell:
 yamllint:
 	hack/check/check-yaml.sh
 
+
+# These are the modules that still contain issues reported by golangci-lint.
+# Once a module is updated to be lint-free, remove from list to catch lint regressions.
+MODULES_NEEDING_LINT_FIX=./cmd/cli/plugin/login ./cmd/cli/plugin/package ./cmd/cli/plugin-admin/builder . ./apis/core ./apis/addonconfigs ./capabilities/client ./addons ./hack/packages/package-tools ./packageclients ./tkg ./tkr ./pinniped-components/post-deploy ./pkg/v1/tkr
+
 go-lint: tools ## Run linting of go source
 	@for i in $(GO_MODULES); do \
 		echo "-- Linting $$i --"; \
-		pushd $${i}; \
-		$(GOLANGCI_LINT) run -v --timeout=10m; \
-		popd; \
+		pushd $${i} > /dev/null; \
+	    if echo ${MODULES_NEEDING_LINT_FIX} | tr ' ' '\n' | grep -q -x $$i; then \
+			$(GOLANGCI_LINT) run -v --timeout=10m || echo "module $$i needs lint fixes"; \
```

Comment:
What do you think about adding a prefix to the printout that would make it more noticeable?  Maybe something people would search for?  Maybe "ERROR: module $$i needs lint fixes".  Or if you don't like using "ERROR", we could use "WARNING", although I don't think people search for that string when looking at those giant CI logs.

Reply:
Oh, and how about adding a different printout if the linter succeeds here, to say that this module should be removed from the `MODULES_NEEDING_LINT_FIX` variable?

Reply:
Good point, that thought came across my mind too, but I reasoned that fixing the module requires a conscious effort so it is quite likely the need to update MODULES_NEEDING_LINT_FIX is not lost on the person making the fix. 
But turns out it is not complicated to introduce another conditional check, so let me fix it.

Reply:
updated